File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String str = s.toLowerCase();
        str = str.replaceAll("[^a-z0-9]", "");

        if (str.isEmpty()) return true;

        StringBuilder reversed = new StringBuilder(str);
        reversed.reverse();
        String rstr = reversed.toString();

        return str.equals(rstr);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.ts
function isPalindrome(s: string): boolean {
    const strClean = s.toLowerCase().replace(/[^a-z0-9]/g, '');
    if (!strClean) {
        return true;
    }
    const reversed = strClean.split('').reverse().join('');
    return strClean === reversed;
}
